src/sql/workbench/contrib/notebook/browser/notebookViews/insertCellsModal.ts

```diff
@@ -0,0 +1,299 @@
+			let optionElement = this._optionsMap[key];
+			if (optionElement.currentValue === true) {
+				const activeView = this._context.getActiveView();
+				const cellToInsert = activeView.getCell(optionElement.optionMetadata.name);
+				if (cellToInsert) {
+					this.onInsert(cellToInsert);
+				}
+			}
+		}
+	}
+
+	public setOptionValue(optionName: string, value: boolean): void {
+		if (this._optionsMap[optionName] !== undefined) {
+			this._optionsMap[optionName].currentValue = value;
+		}
+	}
+
+	public getOptionValue(optionName: string): boolean {
+		return this._optionsMap[optionName]?.currentValue;
+	}
+}
+
+export class InsertCellsModal extends Modal {
+	public viewModel: CellOptionsModel;
+
+	private _submitButton: Button;
+	private _cancelButton: Button;
+	private _optionsMap: { [name: string]: Checkbox } = {};
+	private _maxTitleLength: number = 20;
+
+	constructor(
+		private onInsert: (cell: ICellModel) => void,
+		private _context: NotebookViewsExtension,
+		private _containerRef: ViewContainerRef,
+		private _componentFactoryResolver: ComponentFactoryResolver,
+		@ILogService logService: ILogService,
+		@IThemeService themeService: IThemeService,
+		@ILayoutService layoutService: ILayoutService,
+		@IClipboardService clipboardService: IClipboardService,
+		@IContextKeyService contextKeyService: IContextKeyService,
+		@IAdsTelemetryService telemetryService: IAdsTelemetryService,
+		@ITextResourcePropertiesService textResourcePropertiesService: ITextResourcePropertiesService,
+	) {
+		super(
+			localize("insertCellsModal.title", "Insert cells"),
+			'InsertCellsModal',
+			telemetryService,
+			layoutService,
+			clipboardService,
+			themeService,
+			logService,
+			textResourcePropertiesService,
+			contextKeyService,
+			{ hasErrors: true, hasSpinner: true }
+		);
+
+		const options = this.getOptions();
+		this.viewModel = new CellOptionsModel(options, this.onInsert, this._context);
+	}
+
+	protected renderBody(container: HTMLElement): void {
+		const grid = DOM.$<HTMLDivElement>('div#insert-dialog-cell-grid');
+
+		grid.style.display = 'grid';
+		grid.style.gridTemplateColumns = '1fr 1fr';
+		grid.style.gap = '10px';
+		grid.style.padding = '10px';
+		grid.style.overflowY = 'auto';
+		grid.style.maxHeight = 'calc(100% - 40px)';
+
+		const gridTitle = DOM.$<HTMLHeadElement>('h2.grid-title');
+		gridTitle.title = localize("insertCellsModal.selectCells", "Select cell sources");
+
+		DOM.append(container, grid);
+
+		this.createOptions(grid)
+			.catch((e) => { this.setError(localize("insertCellsModal.thumbnailError", "Error: Unable to generate thumbnails.")); });
+	}
+
+	protected layout(height: number): void {
+		// No-op for now. No need to relayout.
+	}
+
+	private async createOptions(container: HTMLElement): Promise<void> {
+		const activeView = this._context.getActiveView();
+		const cellsAvailableToInsert = activeView.hiddenCells;
+
+		this._themeService.getColorTheme().getColor(editorBackground);
```

Comment:
You don't appear to be doing anything with this

(and colors should generally be handled in a registerThemingParticipant callback so they're updated when the theme changes)